### PR TITLE
[#8814] Update user map for April 2018

### LINKED ISSUE #8814: Update user map for April 2018
To be done after (and similar to) #8755

New entries to consider:

Australia | Curtin University
-- | --
Australia | Charles Sturt University
Australia | Melbourne Business School
Australia | Monash University
Australia | Southern Cross University
Australia | Swinburne
Australia | Swinburne University of Technology
Australia | University of Melbourne
Australia | University of South Australia
Australia | University of Technology Sydney
Belgium | Sint-Lievenscollege Antwerpen
Canada | Cégep de Saint-Jérôme
Canada | École de technologie supérieure
Canada | George Brown College
Canada | Kwantlen Polytechnic University
Canada | Luther College
Canada | Okanagan College
Canada | University of British Columbia
Canada | University of Toronto Scarborough
China | Yi Dian Zi Xun
Colombia | Colegio Nueva Granada
Denmark | IBC
France | université de savoie
France | University of Paris 7
Indonesia | Perbanas Institute
Japan | International University of Japan (IUJ)
Kazakhstan | Nazarbayev University
Korea | Michuhol High School
Malaysia | Universiti Malaysia Perlis
Malaysia | Universiti Tenaga Nasional
Netherlands | Saxion
Netherlands | Saxion University of Applied Sciences
New Zealand | University of Auckland
New Zealand | Victoria University of Wellington
Saudi Arabia | King Fahd University of Petroleum and Minerals
Singapore | Damai Secondary School
Singapore | Jurong Junior College
Singapore | Nanyang Polytechnic
Singapore | Nanyang Technological University
Singapore | Ngee Ann Polytechnic
Singapore | Singapore Institute of Technology
Singapore | Singapore Polytechnic
Sweden | Chalmers University
UK | The Royal Veterinary College
UK | Univeristy of Manchester
UK | University of Bradford
USA | Arroyo High School
USA | Astor School
USA | Belmont University
USA | Chattahoochee High School
USA | DePaul University
USA | Duke University
USA | Florida International University
USA | Floyd College and Career Academy
USA | Fort Lewis College
USA | Grandview High School
USA | Hult International Business School
USA | Kennesaw State University
USA | Lafayette High School
USA | Lake Forest College
USA | Lander University
USA | Marquette University
USA | MassBay Community College
USA | Miami University
USA | Michigan Technological University
USA | Missouri State University
USA | Mountain Home Public School
USA | New Mexico State University
USA | North Carolina State University
USA | North Dakota State University
USA | Northeastern University
USA | Northern Arizona University
USA | Oak Creek High School
USA | Oakland University
USA | Ohio State University
USA | Parker University
USA | Regis University
USA | Rio Hondo Community College
USA | Rochester Institute of Technology
USA | San Diego State University
USA | San Francisco State University
USA | Seven Lakes High School
USA | Southern Oregon University
USA | Southern Oregon University
USA | Spring Lake Park High School
USA | Sutton Elementary School
USA | Syracuse University
USA | Texas A&M University
USA | The University of Texas Rio Grande Valley
USA | University at Albany
USA | University of California, Irvine
USA | University of Chicago
USA | University of Chicago
USA | University of Cincinnati
USA | University of Hawaii
USA | University of Kansas
USA | University of Kentucky
USA | University of Louisiana at Lafayette
USA | University of Mass, Lowell
USA | University of Minnesota
USA | University of North Carolina
USA | University of North Carolina at Wilmington
USA | University of North Florida
USA | University of Oregon
USA | University of South Dakota
USA | University of Southern California
USA | University of Virginia
USA | University of Washington
USA | Villanova University
USA | Violet Fairweather-Baker
USA | Washington and Lee University
USA | Washington State University
USA | Webster University
USA | Western Washington University
USA | Xavier University of Louisiana

### PR DESCRIPTION
**Pull request by cbb692**
Fixes #8814 

Added the list of requested schools to the user Map Data file. I have two things that might require a repull regarding some listings, but for now this should be alright.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (dalessr) [inline .vscode/settings.json]: Unnecessary changes here that should be removed. Since you are using `Visual Studio Code`, maybe you want to include `.vscode` in `.gitignore`. Also, we suggest you use `IntelliJ` or `Eclipse` for development.

File: src/main/webapp/data/userMapData.json [hunk 2]
```diff
@@ -330,6 +338,7 @@
       "Danish Hydraulic Institute",
       "Erhvervsakademiet Lillebælt",
       "Hans Christian Ørsted Gymnasium",
+      "IBC",
```
Comment (dalessr) [inline src/main/webapp/data/userMapData.json]: Can give a full name for this?

File: src/main/webapp/data/userMapData.json [hunk 3]
```diff
@@ -398,7 +407,9 @@
       "SKEMA Business School",
       "Sup'Biotech",
       "Telecom Paris Tech",
-      "University of Lyon"
+      "University of Lyon",
+      "Université de Savoie",
```
Comment (dalessr) [inline src/main/webapp/data/userMapData.json]: The order doesn't seem to be correct here.

File: src/main/webapp/data/userMapData.json [hunk 4]
```diff
@@ -1643,7 +1677,9 @@
       "Northwood University",
       "Norwich Technical High School",
       "Nyack High School",
+      "Oakland University",
       "Oakton Community College",
+      "Oak Creek High School",
```
Comment (dalessr) [inline src/main/webapp/data/userMapData.json]: Should we put this line before `Oakland University`?

File: src/main/webapp/data/userMapData.json [hunk 5]
```diff
@@ -1890,6 +1933,7 @@
       "University of Kansas Medical Center",
       "University of Kentucky",
       "University of Louisiana Monroe",
+      "University of Louisiana at Lafayette",
```
Comment (dalessr) [inline src/main/webapp/data/userMapData.json]: The order is incorrect here.

**Comment by cbb692** [comment]
Ok...other than those two things, I think I got it all finally.

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,4 @@
+.vscode
```
Comment (dalessr) [inline .gitignore]: Sorry I think I didn't explain this clearly. If you are using `vscode`, you can add this to `.gitignore` so that the changes in `.vscode` won't be committed. But you need to remove this from `.gitignore` before creating a PR. However, IDEs such as `IntelliJ IDEA` are still highly recommended instead of `vscode`.

File: src/main/webapp/data/userMapData.json (hunk 2 above)
Comment (dalessr) [inline src/main/webapp/data/userMapData.json]: Are you able to find a full name for this institute?

File: src/main/webapp/data/userMapData.json [hunk 7]
```diff
@@ -717,6 +731,7 @@
       "Rotterdam Business School",
       "Rotterdam School of Management",
       "Rotterdam University of Applied Science",
+      "Saxion",
```
Comment (dalessr) [inline src/main/webapp/data/userMapData.json]: Also, is there a full name for this? Is it different from the one below?

File: src/main/webapp/data/userMapData.json (hunk 4 above)
Comment (dalessr) [reply src/main/webapp/data/userMapData.json]: Ok I think for now it is enough for us to put this entry in front. On a side note, I'm not sure about how you actually compare the two entries, but in my view, even comparing `Oakton...` and `OakCreek...` should put `OakCreek...` in front.

**Comment by cbb692** [comment]
Hey @dalessr , thanks for checking back. Just wanted to go through your notes in a single post to maybe make things easier to read.

**tl;dr I'll fix 1 and 4, 3 can be removed most likely, and 2 needs further input.**

1 [.vscode]) No problem. I cloned to a different computer that runs Eclipse so I should hopefully have those issues fixed tonight. Sorry for the trouble.

2 [IBC]) I mentioned this in a comment above, but the college refers to itself as either "IBC" or "IBC International Business College" (which quells what it actually stands for). So I guess the question is between "IBC", "International Business College", or "IBC International Business College", what do you prefer?

3 [Saxion]) Was on the list twice so I added it in but yea, DuckDuckGo shows only the Applied Sciences one. I can remove "Saxion" assuming you don't see anything different.

4 [Oak Creek]) Ok I can move it in front. I might need to go through the whole list and check for instances of this (where words with spaces are added after even if they would come before strictly based on alphabetical order), but for now I can just move Oak Creek in front.

**Comment by damithc** [comment]
> 2 [IBC]) I mentioned this in a comment above, but the college refers to itself as either "IBC" or "IBC International Business College" (which quells what it actually stands for). So I guess the question is between "IBC", "International Business College", or "IBC International Business College", what do you prefer?

Let's use "International Business College"

**Comment by cbb692** [comment]
@dalessr 
@damithc 

Alright so here's the thing about number 4 above. Updating how we're sorting all the names (i.e. removing all whitespace, comparing, then sorting accordingly) will require a full pass on the UMD file. As an example that hadn't been picked up, "Northwest High School" (line 1672 in my version) will need to be moved between "Northwestern University" and "Northwood University" (1675 and 6, respectively) as "Northwest**H**igh..." will go after "Northwest**e**rn...".

I can either try and pick this up right now as part of this ticket or, in an attempt to only solve 1 issue per...well...issue, we/I/whoever can make a new issue and said individual(s) can pick it up and solve it. 

As always, let me know what works best for you, and I can get it done. Otherwise, the other 3 issues **should** (I pray) be all set.

**Comment by damithc** [comment]
> I can either try and pick this up right now as part of this ticket or, in an attempt to only solve 1 issue per...well...issue, we/I/whoever can make a new issue and said individual(s) can pick it up and solve it.

I prefer if the clean up is done as a separate issue. Things move faster when PRs don't try to do too many things at once :-)

**Comment by cbb692** [comment]
That's pretty much what I expected. In a few hours I'll create a new issue for the UMD pass, but otherwise we can probably close this out as long as the other changes are ok'd
